test/Dockerfile

```diff
@@ -59,4 +60,7 @@ RUN go version
 # Cache modules to speed up test execution
 RUN git clone https://github.com/aws/karpenter.git /karpenter
 WORKDIR /karpenter
-RUN GOPROXY=direct go mod tidy
+
+RUN K8S_VERSION="1.28.x" make toolchain
```

Comment:
Do we need to run the full `make toolchain`? IIRC, we should be fine just defining a subset of dependencies in this Dockerfile and then using those directly.